### PR TITLE
Add logo

### PR DESCRIPTION
**Pull request by oliverviljamaa**
# v0.2.0
## Adds logo

image

Passing the logo link (https://transferwise.com by default):
```js
<PublicNavigation logoLink="https://transferwise.com/borderful" />
```

### CODE REVIEW

File: src/PublicNavigation.js [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
+import PublicNavigationWithoutContent from './publicNavigationWithoutContent';
 
-PublicNavigation.propTypes = {
-  inverse: Types.bool,
-};
-
-PublicNavigation.defaultProps = {
-  inverse: false,
-};
+const PublicNavigation = props => <PublicNavigationWithoutContent {...props} />;
```
Comment (oliverviljamaa) [inline src/PublicNavigation.js]: The wrapper was created because on this level, I'll import a JSON of `items` to show in the nav.

File: src/publicNavigationWithoutContent/logo/Logo.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { PropTypes as Types } from 'prop-types';
+
+const Logo = ({ inverse, link }) => (
+  <div className="navbar-header pull-xs-left">
+    <a
+      href={link || 'https://transferwise.com'}
+      className={`logo-text${inverse ? ' logo-text-inverse' : ''} hidden-md hidden-lg hidden-xl`}
+    >
+      <span hidden>TransferWise</span>
```
Comment (OFFLlNE) [inline src/publicNavigationWithoutContent/logo/Logo.js]: why is this necessary if this is hidden?

File: src/PublicNavigation.js [hunk 3]
```diff
@@ -1,16 +1,7 @@
 import React from 'react';
-import { PropTypes as Types } from 'prop-types';
 
-const PublicNavigation = ({ inverse }) => (
-  <header className={`navbar${inverse ? ' navbar-inverse' : ''}`} />
-);
+import PublicNavigationWithoutContent from './publicNavigationWithoutContent';
```
Comment (adrienDog) [inline src/PublicNavigation.js]: just curious: why the choice of camel case? as opposed to dash (kebab case)
Since lower/upper case is not captured as a change on mac for example

File: src/publicNavigationWithoutContent/PublicNavigationWithoutContent.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  inverse: Types.bool,
+  logoLink: Types.string,
+};
+
+PublicNavigation.defaultProps = {
+  inverse: false,
+  logoLink: '',
+};
+
+export default PublicNavigation;
```
Comment (adrienDog) [inline src/publicNavigationWithoutContent/PublicNavigationWithoutContent.js]: still wanna name it just `PublicNavigation`?
Also, what is this `PublicNavigationWithoutContent`? are you anticipating sthg to come next?
Never seen that so curious to learn

File: src/publicNavigationWithoutContent/logo/Logo.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { PropTypes as Types } from 'prop-types';
+
+const Logo = ({ inverse, link }) => (
+  <div className="navbar-header pull-xs-left">
+    <a
+      href={link || 'https://transferwise.com'}
```
Comment (adrienDog) [inline src/publicNavigationWithoutContent/logo/Logo.js]: could have been a `const DEFAULT_LOGO_LINK = 'https://transferwise.com'` (gives more details to future readers. Also it seems the value is used in two places in this file

File: src/publicNavigationWithoutContent/logo/Logo.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      className={`logo-text${inverse ? ' logo-text-inverse' : ''} hidden-md hidden-lg hidden-xl`}
```
Comment (OFFLlNE) [inline src/publicNavigationWithoutContent/logo/Logo.js]: what exactly is the logo-text class? What do we display on sm and xs then?
Not the actual flag with TransferWise, but just something else? :D

File: src/publicNavigationWithoutContent/logo/Logo.js (hunk 2 above)
Comment (oliverviljamaa) [reply src/publicNavigationWithoutContent/logo/Logo.js]: For accessibility:
[jsx-a11y/anchor-has-content](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md)

Comment (omarduarte) [reply src/publicNavigationWithoutContent/logo/Logo.js]: Screen readers?

File: src/publicNavigationWithoutContent/PublicNavigationWithoutContent.js (hunk 4 above)
Comment (oliverviljamaa) [reply src/publicNavigationWithoutContent/PublicNavigationWithoutContent.js]: Not super happy with the name, but I want to have the items config in the root, and the `withoutContent` object rendering whatever it's given.

File: src/publicNavigationWithoutContent/logo/Logo.js (hunk 6 above)
Comment (oliverviljamaa) [reply src/publicNavigationWithoutContent/logo/Logo.js]: Just `TransferWise`, without the tiny flag on top-left.

image

File: src/PublicNavigation.js (hunk 3 above)
Comment (oliverviljamaa) [reply src/PublicNavigation.js]: I'm following the conventions seen in TransferWise, I personally prefer them to reflect whatever the name of the importable is (in this case, `PublicNavigationWithoutContent` with a capital first), but I haven't seen this, so went with the default.

I've also never seen anyone doing kebab-case, TBF.

If we had a convention, we wouldn't have to capture the changes, as we'd know what to do, though.

Comment (oliverviljamaa) [reply src/PublicNavigation.js]: After looking at these:
https://medium.com/@alexmngn/how-to-better-organize-your-react-applications-2fd3ea1920f1
https://hackernoon.com/the-100-correct-way-to-structure-a-react-app-or-why-theres-no-such-thing-3ede534ef1ed

I'm keen on naming it `PublicNavigationWithoutContent`, no good reason for the import to be different based on whether it's in the same dir or not.

Comment (adrienDog) [reply src/PublicNavigation.js]: Yes if we want them to be named the same let's go all the way :)
